odoo/addons/test_testing_utilities/tests/test_methods.py

```diff
@@ -28,3 +38,44 @@ def test_assertRecordValues(self):
             self.assertRecordValues(records, [X1, Y3])
         with self.assertRaises(AssertionError):
             self.assertRecordValues(records, [Y3, X1])
+
+    def test_assertRaises_rollbacks(self):
+        self.env.cr.execute("SET LOCAL test_testing_utilities.a_flag = ''")
+        with self.assertRaises(CustomError):
+            self.env.cr.execute("SET LOCAL test_testing_utilities.a_flag = 'yes'")
+            raise CustomError
+
+        self.env.cr.execute("SHOW test_testing_utilities.a_flag")
+        self.assertEqual(self.env.cr.fetchone(), ('',))
+
+    def test_assertRaises_error(self):
+        """Checks that an exception raised during the *setup* of assertRaises
+        bubbles up correctly.
+        """
+        # ensure we catch the error with the "base" method to avoid any interference
+        with TestCase.assertRaises(self, CustomError),\
+             mock.patch.object(BaseCursor, 'flush', side_effect=CustomError):
+
```

Comment:
Shouldn't we mock outside the first `assertRaises`?  I noticed this test and the next one are not consistent on where monkey-patches are made.
```suggestion
        with mock.patch.object(BaseCursor, 'flush', side_effect=CustomError), \
             TestCase.assertRaises(self, CustomError):
```

Reply:
Maybe. I think I did that on purpose initially but I can't remember why. It might have gotten lost in the shuffle as I found better ways to do things.

Not sure which one makes most sense as the inner-most context, so I guess this makes as much sense as the reverse, it's not like there's a dependency between the two (hopefully).